script.js

```diff
@@ -1,12 +1,208 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+//Variables
+// Div for showing episodes or movies
+const showEpisodseRow = document.querySelector(".row");
+// Search input
+const searchUser = document.querySelector("#searchUser");
+// Show the number of episodes based on search
+const searchNumResult = document.querySelector(".epi-num");
+// Select related to shows
+const selectShows = document.querySelector("#select-shows");
+// Select related to episodes
+const selectEpisodes = document.querySelector("#select-episodes");
+let allShows;
+let allEpisodes;
+
+//EventListeners
+// display episodes on load
+document.addEventListener("DOMContentLoaded", setup);
+// Search specific Show based on select
+selectShows.addEventListener("change", selectShow);
+// Search specific episode based on select
+selectEpisodes.addEventListener("change", selectEpisode);
+// Search episodes
+searchUser.addEventListener("keyup", searchMovie);
+
+//Functions
+// Get all shows
+async function setup() {
+  //get all shows with fetch
+  allShows = await getAllAvailablShows();
+  // Make card for all shows and create options for select
+  makeCardForShows(allShows);
+}
+
+//Get all shows via fetch
+async function getAllAvailablShows() {
+  // await response of the fetch call
+  const response = await fetch("https://api.tvmaze.com/shows");
+  // only proceed once its resolved
+  const data = await response.json();
+  // only proceed once second promise is resolved
+  return data;
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// Make card for all shows
+function makeCardForShows(allShows) {
+  searchUser.disabled = true;
+  allShows.sort((a, b) =>
+    a.name.toLowerCase().localeCompare(b.name.toLowerCase())
+  );
+  let output = "";
+  allShows.forEach((show) => {
+    output += `
+      <div class="col" style="height: 550px">
```

Comment:
This works, but can you try using the DOM JavaScript APIs rather than encoding HTML as strings? (In real life, either is probably fine, but part of the point of this project was to practice using the DOM APIs :))